core/src/main/java/org/apache/calcite/sql/SqlAggFunction.java

```diff
@@ -38,6 +38,7 @@
   private final boolean requiresOrder;
   private final boolean requiresOver;
   private final Optionality requiresGroupOrder;
+  private final Boolean ignoreNulls;
```

Comment:
Can we use boolean here?

Reply:
OK, I see you have null check below, that's why it is Boolean type.

Reply:
You are right. Maybe I should leave some comments here to avoid confusing.

Reply:
A couple of 'data modeling' issues:
* Whether to respect or ignore nulls seems, to me, to be a property of the call, not the operator. When you are calling a function and asking it to ignore nulls, you are not creating a new operator, but are creating a new call to an existing operator.
* I would love it if we can change `ignoreNulls` from a nullable `Boolean` to a not-null `boolean`. (I hate null values in Java.) Since the default is `IGNORE NULLS`, do we really care whether they wrote `LAST_VALUE` or `LAST_VALUE IGNORE NULLS`?

Reply:
Thanks @julianhyde. Actually the default is `RESPECT NULLS`(Oracle: https://docs.oracle.com/cd/E11882_01/server.112/e41084/functions082.htm#SQLRF00652). 
But your point makes sense. I would like to try it.

Reply:
@julianhyde I have some question:

Reply:
That does sound like an attractive option. The other ways to represent RESPECT NULLS in SqlNode - namely to modify the operator, and to extend SqlCall - have significant problems.

Let's have a discussion in the JIRA issue, and agree a design. I'm sorry that you wrote a lot of code before we had the design discussion.

Reply:
It will be great if we can have more discussion.